integration_tests/src/main/python/array_test.py

```diff
@@ -284,6 +284,33 @@ def test_array_max(data_gen):
                 'array_max(a)'),
             conf=no_nans_conf)
 
+
+@pytest.mark.parametrize('data_gen', orderable_gens + nested_gens_sample, ids=idfn)
+def test_array_repeat_with_count_column(data_gen):
+    cnt_gen = IntegerGen(min_val=-5, max_val=5, special_cases=[])
+    cnt_not_null_gen = IntegerGen(min_val=-5, max_val=5, special_cases=[], nullable=False)
+    gen = StructGen(
+        [('elem', data_gen), ('cnt', cnt_gen), ('cnt_nn', cnt_not_null_gen)], nullable=False)
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: gen_df(spark, gen).selectExpr(
+            'array_repeat(elem, cnt)',
+            'array_repeat(elem, cnt_nn)',
+            'array_repeat("abc", cnt)'))
+
+
+@pytest.mark.parametrize('data_gen', orderable_gens + nested_gens_sample, ids=idfn)
+def test_array_repeat_with_count_scalar(data_gen):
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: unary_op_df(spark, data_gen).selectExpr(
+            'array_repeat(a, 3)',
+            'array_repeat(a, 1)',
+            'array_repeat(a, 0)',
+            'array_repeat(a, -2)',
+            'array_repeat("abc", 2)',
```

Comment:
Because both parameters are scalars the call is probably replaced with a single scalar array at compile time. Have you checked that the test plan is covering what you want/expect?

Reply:
```
GpuColumnarToRow false
+- GpuProject [gpuarrayrepeat(a#16, 3) AS array_repeat(a, 3)#18, gpuarrayrepeat(a#16, 1) AS array_repeat(a, 1)#19, gpuarrayrepeat(a#16, 0) AS array_repeat(a, 0)#20, gpuarrayrepeat(a#16, -2) AS array_repeat(a, -2)#21, [abc,abc] AS array_repeat(abc, 2)#22, [] AS array_repeat(abc, 0)#23, [] AS array_repeat(abc, -1)#24]
   +- GpuRowToColumnar targetsize(104857600)
      +- *(1) Scan ExistingRDD[a#16]
```
From the dumped plan, the optimizing of constant exprs doesn't seem to happen.